src/targets/gpu/hip.cpp

```diff
@@ -304,9 +304,17 @@ argument get_preallocation(context& ctx, const std::string& id)
 
 void gpu_fill(context& ctx, const argument& dst, int value)
 {
-    // TODO: Handle non-packed tensor when value is not 0
-    assert(dst.get_shape().packed() and value == 0);
-    hip_async_memset(ctx, dst, value);
+    if(dst.get_sub_objects().empty())
+    {
+        // TODO: Handle non-packed tensor when value is not 0
+        assert(dst.get_shape().packed() and value == 0);
```

Comment:
Just assert here or do we want to add a throw if we hit that case but this isn't handled?

Reply:
Currently we are not calling this with a non-packed tensor which is why its an assert.